### PR TITLE
unify: init at 0.5

### PR DESCRIPTION
**Pull request by FlorianFranzen**
###### Motivation for this change

unify was missing from nixpkgs, but widely used.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/unify/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchFromGitHub, untokenize }:
```
Comment (mweinelt) [inline pkgs/development/python-modules/unify/default.nix:1]: Expand this to one input per line, this makes diffing and merging easier.

```suggestion
{ lib
, buildPythonPackage
, fetchFromGitHub
, untokenize
}:
```

File: pkgs/development/python-modules/unify/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "myint";
+    repo = "unify";
+    rev = "v${version}";
+    sha256 = "1l6xxygaigacsxf0g5f7w5gpqha1ava6mcns81kqqy6vw91pyrbi";
+  };
+
+  propagatedBuildInputs = [ untokenize ];
+
+  meta = with lib; {
+    description = "Modifies strings to all use the same quote where possible";
```
Comment (mweinelt) [inline pkgs/development/python-modules/unify/default.nix:18]:
```suggestion
    description = "Modifies strings to all use the same quote where possible";
    homepage = "https://github.com/myint/unify";
```

Adding the homepage makes it easier to quickly navigate to the upstream codebase.

File: pkgs/development/python-modules/unify/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  # PyPi release is missing tests (see https://github.com/myint/unify/pull/18)
+  src = fetchFromGitHub {
+    owner = "myint";
+    repo = "unify";
+    rev = "v${version}";
+    sha256 = "1l6xxygaigacsxf0g5f7w5gpqha1ava6mcns81kqqy6vw91pyrbi";
+  };
+
+  propagatedBuildInputs = [ untokenize ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/unify/default.nix:15]:
```suggestion
  propagatedBuildInputs = [ untokenize ];
  
  checkPhase = ''
    ${python.interpreter} -m unittest discover
  '';
```

Explicitly call the test suite and don't rely on `python setup.py test`, which is deprecated.

Requires you to import `python` in the function signature.

File: pkgs/development/python-modules/untokenize/default.nix [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{ lib, buildPythonPackage, fetchPypi }:
```
Comment (mweinelt) [inline pkgs/development/python-modules/untokenize/default.nix:1]: Same as above.

```suggestion
{ lib
, buildPythonPackage
, fetchPypi
}:
```

File: pkgs/development/python-modules/untokenize/default.nix [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+buildPythonPackage rec {
+  pname = "untokenize";
+  version = "0.1.1";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3865dbbbb8efb4bb5eaa72f1be7f3e0be00ea8b7f125c69cbd1f5fda926f37a2";
+  };
+
+  meta = with lib; {
+    description = "Transforms tokens into original source code while preserving whitespace";
```
Comment (mweinelt) [inline pkgs/development/python-modules/untokenize/default.nix:12]:
```suggestion
    description = "Transforms tokens into original source code while preserving whitespace";
    homepage = "https://github.com/myint/untokenize";
```

Same as above.

File: pkgs/development/python-modules/untokenize/default.nix [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "untokenize";
+  version = "0.1.1";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3865dbbbb8efb4bb5eaa72f1be7f3e0be00ea8b7f125c69cbd1f5fda926f37a2";
+  };
+
```
Comment (mweinelt) [inline pkgs/development/python-modules/untokenize/default.nix:10]: Add an explicit `checkPhase` as above.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120085` at c9f41f57 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.unify</li>
    <li>python38Packages.untokenize</li>
    <li>python39Packages.unify</li>
    <li>python39Packages.untokenize</li>
  </ul>
</details>

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/unify/default.nix (hunk 3 above)
Comment (FlorianFranzen) [reply pkgs/development/python-modules/unify/default.nix:15]: Thanks, I was just looking this up, as I was struggling to understand how `setuptools` was able to find the testsuite without it being specified. I guess `setuptoolsCheckHoock` should then be updated soon as well.

File: pkgs/top-level/python-packages.nix [hunk 7]
```diff
@@ -8733,6 +8733,8 @@ in {
 
   unifi = callPackage ../development/python-modules/unifi { };
 
+  unify = callPackage ../development/python-modules/unify { };
```
[the remark this answers is not in the document]
Comment (FlorianFranzen) [reply pkgs/top-level/python-packages.nix:8736]: I added a general entry point under the default python3 as well.

**Comment by FlorianFranzen** [comment]
@mweinelt Thank your for the feedback. I fixed all the issue as requested.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**